core/src/toga/app.py

```diff
@@ -520,11 +520,60 @@ def current_window(self):
         """Return the currently active content window."""
         return self._impl.current_window().interface
 
+    @property
+    def maximized(self):
+        """Returns True if any of the app's windows are maximized."""
+        return any(window.maximized for window in self.windows)
+
+    @maximized.setter
+    def maximized(self, maximize: bool):
+        if maximize:
+            for window in self.windows:
+                window.maximized = True
+        else:
+            for window in self.windows:
+                window.maximized = False
+
+    @property
+    def minimized(self):
+        """Returns True only if all of the app's windows are minimized."""
+        return all(window.minimized for window in self.windows)
+
+    @minimized.setter
+    def minimized(self, minimize: bool):
+        if minimize:
+            for window in self.windows:
+                window.minimized = True
+        else:
+            for window in self.windows:
+                window.minimized = False
+
+    @property
+    def full_screen(self):
+        """Return True if any of the app's window are in full screen."""
+        return any(window.full_screen for window in self.windows)
+
+    @full_screen.setter
+    def full_screen(self, full_screen):
+        if full_screen:
+            for window in self.windows:
+                window.full_screen = True
+        else:
+            for window in self.windows:
+                window.full_screen = False
+
+    def set_normal_screen(self):
+        """Sets the app's windows in normal state"""
+        for window in self.windows:
+            window.set_normal_screen()
+
+    # For backwards compatibility in core API
```

Comment:
Backwards compatibility that is only visible with a code comment isn't really backwards compatibility. 

This should be causing a DeprecationWarning, and should then be implemented in terms of the new API, rather than maintaining a parallel implementation.

Reply:
I want to know why it is needed *at all*. As an API entry point, it's completely inconsistent with all the other methods - there's no "inverse", and it's a method, not a property. Why isn't `maximized=False` (or `minimized=False`) sufficient? Why is an explicit method needed for "normal" state? Or, if there *is* a need for a method - why isn't it `set_window_state()`, rather than a *normal* specific method?

Reply:
I mean there might be situations where the app developer doesn't care about what state a particular window is in and just wants the window to not remain in `maximize`, `minimize` or `full_screen` state. 

Making `normal=True/False` as a property doesn't make much sense as doing `normal=False` doesn't make any logical sense. 

Hence, there is a method `set_normal_screen()`.

Reply:
Sure - but then, why not have `set_window_state()`? That means you have an explicit API to handle *all* window states, rather than a single method that stands out as a different pattern to all the others. Yes, `window.set_window_state(WindowState.MAXIMIZED)` is exactly the same as `window.maximized=True`... but that's a documentation issue.

Reply:
@freakboy3742 I agree we can have a `set_window_state()` as a single method. Do you want it to become a core API? If so, please confirm, so that I will change it in the next commit.

Reply:
Yes, I agree a core-level API is called for; however, to be consistent, it should be a `window_state` property with a getter and setter, not a `set_window_state()` method. 

I'd also suggest that the property is `state` on the Window class, and `window_state` on the App class.

Note that this doesn't change the *backend* API, which only requires the get/set_window_state() method (as all the core APIs can be expressed based on that single underlying API).

As an aside - this sort of back-and-forth is why writing up finalising design tickets like #1857 *before* implementing a feature is important. The discussion on #1857 had lots of ideas, but it didn't really get to a final concrete approved design. Rather than go through lots of code churn where the comments are 50% code style and 50% API design questions, it's better to sort out a complete design *first*.